src/react-components/avatar-editor.js

```diff
@@ -113,42 +160,18 @@ class AvatarEditor extends Component {
     e.preventDefault();
 
     if (this.inputFiles.glb && this.inputFiles.glb instanceof File) {
-      const gltfLoader = new THREE.GLTFLoader();
+      const gltfLoader = new THREE.GLTFLoader().register(
+        parser => new GLTFBinarySplitterPlugin(parser, this.inputFiles)
+      );
       const gltfUrl = URL.createObjectURL(this.inputFiles.glb);
       const onProgress = console.log;
-      const parser = await new Promise((resolve, reject) =>
-        gltfLoader.createParser(gltfUrl, resolve, onProgress, reject)
-      );
-      URL.revokeObjectURL(gltfUrl);
 
-      const { body } = parser.extensions.KHR_binary_glTF;
-      const content = JSON.stringify(ensureAvatarMaterial(parser.json));
-      // Inject hubs components on upload. Used to create base avatar
-      // const gltf = parser.json;
-      // Object.assign(gltf.scenes[0], {
-      //   extensions: {
-      //     MOZ_hubs_components: {
-      //       "loop-animation": {
-      //         clip: "idle_eyes"
-      //       }
-      //     }
-      //   }
-      // });
-      // Object.assign(gltf.nodes.find(n => n.name === "Head"), {
-      //   extensions: {
-      //     MOZ_hubs_components: {
-      //       "scale-audio-feedback": ""
-      //     }
-      //   }
-      // });
-      // content = JSON.stringify(gltf);
-
-      this.inputFiles.gltf = new File([content], "file.gltf", {
-        type: "model/gltf"
-      });
-      this.inputFiles.bin = new File([body], "file.bin", {
-        type: "application/octet-stream"
+      await new Promise((resolve, reject) => {
+        // This GLTFBinarySplitterPlugin saves gltf and bin in inputFiles.gltf/bin
```

Comment:
Something about this feels a bit weird but I don't really have a much better idea. I maybe would have had the plugin just store it on itself rather than have it passed in and then done something like this after load:

```js
this.inputFiles.gltf = binarySplitterPlugin.gltf;
this.inputFiles.bin = binarySplitterPlugin.bin;
```

But not sure that is really much better.

Reply:
How about this?

```
await new Promise((resolve, reject) => {
  // GLTFBinarySplitterPlugin saves gltf and bin in gltf.files
  gltfLoader.load(gltfUrl, result => {
    this.inputFiles.gltf = result.files.gltf;
    this.inputFiles.bin = result.files.bin;
  }, onProgress, reject);
});
```